core/src/jsMain/kotlin/Transaction.kt

```diff
@@ -23,6 +24,26 @@ public open class Transaction internal constructor(
         }
     }
 
+    internal suspend fun awaitFailure() {
+        transaction.onNextEvent("complete", "abort", "error") { event ->
+            when (event.type) {
+                "abort" -> Unit
+                "error" -> Unit
+                else -> Unit
+            }
+        }
+    }
+    internal fun abort() {
+        transaction.abort()
+    }
+
+    internal fun commit() {
+        // Check if function exists before calling it.
+        if (transaction.unsafeCast<Json>()["commit"] != undefined) {
```

Comment:
I'm not sure what the standard approach is for this (as JavaScript is outside my wheelhouse) but I've often seen these kind of checks written as:

```suggestion
        if (typeof transaction.commit === 'function') {
```

..if the above approach is functionally the same, I think I'd prefer it from a readability perspective. 🤷 

Let me know your thoughts / rationale behind the casting to `Json`.

Reply:
Thanks! The prosed solution looks much better. I'll check if it works as expected, then prepare update.

Reply:
@twyatt I've tried with `jsTypeOf(transaction::commit) === "function"` but this returns true, no matter if the commit function is implement in the web browser or if it's not.

I can use the following code:
```kotlin
val transaction = transaction
if (js("typeof transaction.commit === 'function'"))
```

but this code is as ugly as the code with unsafeCast ;) 

I don't have much preference, so I'm open for your opinion. Maybe you see another solution for this problem?

Reply:
There is also:
```kotlin
jsTypeOf(transaction.unsafeCast<Json>()["commit"]) === "function"
```

 but this is similar to 
```kotlin
transaction.unsafeCast<Json>()["commit"] != undefined
```

Reply:
Yet another solution that I found, but still a little bit hacky:

```kotlin
private inline fun <T> functionExists(obj: T, function: T.() -> KFunction<*>): Boolean = 
    jsTypeOf(obj.unsafeCast<Json>()[obj.function().name]) === "function"
```

Then:

```kotlin
if (functionExists(transaction) { ::commit }) {
    transaction.commit()
}
```

Reply:
> I've tried with `jsTypeOf(transaction::commit) === "function"` but this returns true, no matter if the commit function is implement in the web browser or if it's not. Even tho this looks the best, it doesn't work :/

Oh bummer. Thanks for giving it a try. Let me experiment a little over the weekend and get back to you.

Reply:
@jacek-marchwicki how are you testing when `commit` isn't supported? It looks like all modern browsers supports it; are you downloading an older version of one of the browsers?

<img width="450" alt="Screenshot 2024-11-09 at 11 51 15 PM" src="https://github.com/user-attachments/assets/fa12b35b-f4b7-4ec5-821d-33aca7a9012c">

Reply:
I'm not sure how to test this (with `commit` specifically), but if you have a moment, maybe one last thing to try:

> [!NOTE]
> This is the way that the [coroutines library is checking](https://github.com/Kotlin/kotlinx.coroutines/blob/6c6df2b850382887462eeaf51f21f58bd982491d/kotlinx-coroutines-core/js/src/CoroutineContext.kt#L15) for feature support.

```kotlin
if (jsTypeOf(transaction.asDynamic().commit) != "undefined") {
    transaction.commit()
}
```

or...

```kotlin
if (jsTypeOf(transaction.asDynamic().commit) === "function") {
    transaction.commit()
}
```

Reply:
I'm testing just by not using "commit" name, but temporarily renaming the method to something else that doesn't exist. I.e. "commit2".

Reply:
Btw, I decided to do the check for "commit" method, because even tho it is supported in the newest web browsers, it was introduced later than other features: 

<img width="756" alt="Screenshot 2024-11-10 at 22 59 33" src="https://github.com/user-attachments/assets/f4f99230-2c64-40b0-b254-d304c0856da7">